src/core/thread.cc

```diff
@@ -1,3 +1,8 @@
+// If _FORTIFY_SOURCE is defined then longjmp ends up using longjmp_chk
+// which asserts that you're jumping to the same stack. However, here we
+// are intentionally switching stacks when longjmp'ing, so undefine this
+// option to always use normal longjmp.
+#undef _FORTIFY_SOURCE
```

Comment:
Excellent. Not only does this do the undef in the right place - in the code and not in some unintelligible cmake file, it also reminds the reader that we disabled FORTIFY_SOURCE on this entire source file and if this file grows and this is considered too broad, or maybe should move just jmp_buf_link with its longjmp calls to a separate file.